id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java

```diff
@@ -1,700 +1,709 @@
+			throws IdRepoAppException {
+		return uinRepo.findByUinHash(id);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see io.mosip.kernel.core.idrepo.spi.IdRepoService#updateIdentity(java.lang.
+	 * Object, java.lang.String)
+	 */
+	@Override
+	public Uin updateIdentity(IdRequestDTO request, String uin) throws IdRepoAppException {
+		AnonymousProfileDto anonymousProfileDto = new AnonymousProfileDto();
+		anonymousProfileDto.setRegId(request.getRequest().getRegistrationId());
+		int modResult = securityManager.getSaltKeyForId(uin);
+		String hashSalt = uinHashSaltRepo.retrieveSaltById(modResult);
+		String uinHashwithSalt = securityManager.hashwithSalt(uin.getBytes(), hashSalt.getBytes());
+		String uinHash = modResult + SPLITTER + uinHashwithSalt;
+		try {
+			Uin uinObject = retrieveIdentity(uinHash, IdType.UIN, null, null);
+			anonymousProfileDto.setOldCbeff(uinHash,
+					!anonymousProfileDto.isOldCbeffPresent() && Objects.nonNull(uinObject.getBiometrics())
+							&& !uinObject.getBiometrics().isEmpty()
+									? uinObject.getBiometrics().get(uinObject.getBiometrics().size() - 1).getBioFileId()
+									: null);
+			uinObject.setRegId(request.getRequest().getRegistrationId());
+			if (Objects.nonNull(request.getRequest().getStatus())
+					&& !StringUtils.equals(uinObject.getStatusCode(), request.getRequest().getStatus())) {
+				uinObject.setStatusCode(request.getRequest().getStatus());
+				uinObject.setUpdatedBy(IdRepoSecurityManager.getUser());
+				uinObject.setUpdatedDateTime(DateUtils.getUTCCurrentDateTime());
+			}
+			if (Objects.nonNull(request.getRequest()) && Objects.nonNull(request.getRequest().getIdentity())) {
+				RequestDTO requestDTO = request.getRequest();
+				Configuration configuration = Configuration.builder().jsonProvider(new JacksonJsonProvider())
+						.mappingProvider(new JacksonMappingProvider()).build();
+				DocumentContext inputData = JsonPath.using(configuration).parse(requestDTO.getIdentity());
+				DocumentContext dbData = JsonPath.using(configuration).parse(new String(uinObject.getUinData()));
+				anonymousProfileDto.setOldUinData(dbData.jsonString().getBytes());
+				JSONCompareResult comparisonResult = JSONCompare.compareJSON(inputData.jsonString(),
+						dbData.jsonString(), JSONCompareMode.LENIENT);
+
+				if (comparisonResult.failed()) {
+					updateIdentityObject(inputData, dbData, comparisonResult);
+					uinObject.setUinData(convertToBytes(convertToObject(dbData.jsonString().getBytes(), Map.class)));
+					uinObject.setUinDataHash(securityManager.hash(uinObject.getUinData()));
+					uinObject.setUpdatedBy(IdRepoSecurityManager.getUser());
+					uinObject.setUpdatedDateTime(DateUtils.getUTCCurrentDateTime());
+				}
+
+				if (Objects.nonNull(requestDTO.getDocuments()) && !requestDTO.getDocuments().isEmpty()) {
+					anonymousProfileDto.setNewCbeff(uinObject.getUinHash(),
+							!anonymousProfileDto.isNewCbeffPresent() && Objects.nonNull(uinObject.getBiometrics())
```

Comment:
This piece of code is confusing, can we restructure this ?

Reply:
Changes done accordingly

Reply:
Resolved